File: pipeline/utils/env.py
from dotenv import load_dotenv
import os


def find_env(var_name: str) -> str:
    load_dotenv()
    value = os.getenv(var_name)
    if value is None:
        raise KeyError(f"Required environment variable '{var_name}' is not set.")
    return value


def find_env_dir(var_name: str) -> str:
    root_path = find_env("ROOT_DIR")
    dir_path = find_env(var_name)
    full_path = os.path.join(root_path, dir_path)
    os.makedirs(full_path, exist_ok=True)

    return full_path

File: R/utils/env.R
library(dotenv)
library(rprojroot)

.root <- rprojroot::find_root(rprojroot::has_file("pixi.toml"))
.env_path <- file.path(.root, ".env")

get_env <- function(key) {
  dotenv::load_dot_env(file = .env_path)

  path <- Sys.getenv(key, unset = NA)
  if (is.na(path) || path == "") {
    stop(paste("Environment variable not set:", key))
  }
  path
}

get_env_dir <- function(key) {
  root_path <- get_env("ROOT_DIR")
  path <- get_env(key)
  full_path <- file.path(root_path, path)

  if (!dir.exists(full_path)) {
    dir.create(full_path, recursive = TRUE)
  }
  full_path
}
